packages/contracts/scripts/tasks/auctionRaffle/ethereumTasks.ts

```diff
@@ -0,0 +1,40 @@
+import { task, types } from 'hardhat/config'
+import { BigNumber, utils } from 'ethers'
+import { EthereumProvider } from 'hardhat/types'
+
+interface Result {
+  blockHash: string,
+  randomNumber: BigNumber,
+}
+
+task('generate-random-numbers', 'Generates random numbers')
+  .addParam('blocks', 'Array of block numbers', undefined, types.json)
+  .addParam('secret', 'Secret number', undefined, types.int)
+  .setAction(async ({ blocks, secret }: { blocks: number[], secret: number }, hre) => {
```

Comment:
JS type `number` would not work with a 256-bit secret. I created a special `bytes23` type for it. Can you review my changes?

Reply:
Changes look good to me. Just updated readme with correct secret type.